### ISSUE TITLE
Invite collaborator

### ISSUE THREAD

**Issue by lalomartins** [gogits/gogs#1308]
I'd like to share my repo with someone who doesn't (yet) have an account in my gogs instance.

One possible implementation is asking for an email, then sending an invite to that email, while remembering in the db that any future users registering with that email will automatically be added to that repo (which could accumulate).

Another possibility is adding github users. If gogs has OAuth login with github, you could enter a github username, then send the repo link to the person, and they'd just have to login/register using github.

(Technically, that could even be done by creating the user when they're first added to a repo; since the system knows this user has neither password or OAuth token, it's a “potential” user which could be excluded from certain features until it's realized by actually logging in, either with email or github.)

Related: #837 for inviting users from other gogs instances. (In fact, a much cheaper, partial version of #837 could also be implemented if gogs instances have an OAuth-like system to log in with each other.)

**Comment by matt-h** [comment]
Gogs does have OAuth login with github check the [oauth.github] section of the config https://github.com/gogits/gogs/blob/master/conf/app.ini
It also has [oauth.google] if you'd rather use Google.

**Comment by lalomartins** [comment]
You're right, of course! I forgot about that because I didn't set it up on mine :smile_cat: so that's an even easier avenue for implementing this feature.

**Comment by lalomartins** [comment]
apropos: #1103

**Comment by mgracer48** [comment]
+100
